src/coreclr/jit/gentree.cpp

```diff
@@ -25071,6 +25083,44 @@ bool GenTreeHWIntrinsic::OperIsMemoryStoreOrBarrier() const
     }
 }
 
+//------------------------------------------------------------------------
+// OperIsEmbBroadcastCompatible: Checks if the intrinsic is a embedded broadcast compatible inintrsic.
+//
+// Return Value:
+//     true if the intrinsic node lowering instruction is embedded broadcast compatible.
+//
+bool GenTreeHWIntrinsic::OperIsEmbBroadcastCompatible() const
+{
+    return HWIntrinsicInfo::IsEmbBroadcastCompatible(GetHWIntrinsicId());
+}
+
+//------------------------------------------------------------------------
+// OperIsBroadcastScalar: Is this HWIntrinsic a broadcast node from scalar.
+//
+// Return Value:
+//    Whether "this" is a broadcast node from scalar.
+//
+bool GenTreeHWIntrinsic::OperIsBroadcastScalar() const
+{
+#if defined(TARGET_XARCH)
+    NamedIntrinsic intrinsicId = GetHWIntrinsicId();
+    switch (intrinsicId)
+    {
+        case NI_AVX2_BroadcastScalarToVector128:
+        case NI_AVX2_BroadcastScalarToVector256:
+        case NI_AVX_BroadcastScalarToVector128:
+        case NI_AVX_BroadcastScalarToVector256:
+        case NI_SSE3_MoveAndDuplicate:
+        case NI_AVX512F_BroadcastScalarToVector512:
```

Comment:
I'm fine with it being handled in a followup, but this list should include `AVX512BW`_BroadcastScalarToVector512` given the above: https://github.com/dotnet/runtime/pull/84821/files#r1198307128

Would be nice to have a tracking issue for it.

Reply:
Will open a tracking issue tomorrow.

Reply:
Might even be good for this to just be a flag on the intrinsic given the sparseness of the 6-7 items we have available here.

Reply:
did you open the issue for the follow-up work?

Reply:
tracked by [issue#86986](https://github.com/dotnet/runtime/issues/86986), sorry for the delay.